File: bicitaxi/lib/features/rides/repository/ride_repository.dart
import '../models/ride.dart';

/// Abstract repository for ride operations.
/// TODO: Replace with Firebase / Firestore implementation.
abstract class RideRepository {
  /// Creates a new ride.
  Future<Ride> create(Ride ride);

  /// Updates an existing ride.
  Future<void> update(Ride ride);

  /// Gets ride history for a specific client.
  Future<List<Ride>> historyForClient(String clientId);

  /// Gets ride history for a specific driver.
  Future<List<Ride>> historyForDriver(String driverId);

  /// Gets a single ride by ID.
  Future<Ride?> getById(String id);

  /// Gets all pending rides (for drivers to see available requests).
  Future<List<Ride>> getPendingRides();
}

File: ios/bicitaxi-conductor/bicitaxi-conductor/RideRepository.swift
//
//  RideRepository.swift
//  bicitaxi-conductor
//
//  Repository protocol for ride data operations
//

import Foundation

/// Protocol for ride data operations
/// TODO: Replace with a real backend implementation (e.g. Firebase, REST API).
protocol RideRepository: Sendable {
    /// Create a new ride
    func create(_ ride: Ride) async throws -> Ride
    
    /// Update an existing ride
    func update(_ ride: Ride) async throws
    
    /// Get ride history for a client
    func historyForClient(id: String) async throws -> [Ride]
    
    /// Get ride history for a driver
    func historyForDriver(id: String) async throws -> [Ride]
    
    /// Get all pending rides (for drivers)
    func pendingRides() async throws -> [Ride]
    
    /// Get a ride by ID
    func getRide(id: String) async throws -> Ride?
}
